code/_macros.dm

```diff
@@ -20,6 +20,7 @@
 #define CanPhysicallyInteractWith(user, target) CanInteractWith(user, target, GLOB.physical_state)
 
 #define QDEL_NULL_LIST(x) if(x) { for(var/y in x) { qdel(y) }}; if(x) {x.Cut(); x = null } // Second x check to handle items that LAZYREMOVE on qdel.
+#define QDEL_NULL_MAP(x) if(x) { for(var/y in x) { if(x[y]) {qdel(x[y]); x[y] = null }}}; if(x) {x.Cut(); x = null } // Second x check to handle items that LAZYREMOVE on qdel.
```

Comment:
This already exists as `QDEL_LIST_ASSOC_VAL` 
I'm not sure why `QDEL_NULL_LIST` is alone in this file, figure it should be with the rest. 
I don't believe nulling the list in either case actually makes a difference since the references are removed by Cut, so we should probably be actually using `QDEL_LIST` which is upstream's

Reply:
Used the QDEL_LIST_ASSOC_VAL and moved the QDEL_NULL_LIST along with other qdels.